modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -264,6 +264,21 @@ def from_pandas(cls, df, data_cls):
     def from_arrow(cls, at, data_cls):
         return cls(data_cls.from_arrow(at))
 
+    # Dataframe exchange protocol
+
+    def to_dataframe(self, nan_as_null: bool = False, allow_copy: bool = True) -> dict:
+        raise NotImplementedError(
+            "The selected execution does not implement the DataFrame exchange protocol yet."
+        )
+
+    @classmethod
+    def from_dataframe(cls, df, data_cls):
```

Comment:
why the signature is this - why we require both `df` and `data_cls`?.. I could use a docstring explaining the parameters...

Reply:
The docstring is in the base class's method. Why we require `data_cls` is to refer to modin_frame. Look at from_pandas implementation as an example.
https://github.com/modin-project/modin/blob/02e4ec90af3fa9b3cbaa844182f71ee87a281952/modin/core/storage_formats/pandas/query_compiler.py#L261